.github/actions/frontend-shared-version/index.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const core = require('@actions/core');
+const util = require('util');
+
+const exec = util.promisify(require('child_process').exec);
+
+/**
+ * Returns true is there is a non-zero length diff between the last published version
+ * and master within the frontend-shared/ folder.
+ */
+async function haveChangesOccurred() {
+  const parentPackage = core.getInput('parent-package');
+  const versionResult = await exec(`npm view ${parentPackage} version`);
+  const version = versionResult.stdout.trim();
+  const diffResult = await exec(`git diff v${version}..master frontend-shared/`)
+  return diffResult.stdout.length > 0;
+}
+
+/**
+ * Increment the minor part of a `MAJOR.MINOR.PATCH` semver version.
+ */
+function bumpMinorVersion(version) {
+  const parts = version.split('.')
+  if (parts.length !== 3) {
+    throw new Error(`${version} is not a valid MAJOR.MINOR.PATCH version`);
+  }
+  const newMinorVersion = parseInt(parts[1]) + 1;
+  return `${parts[0]}.${newMinorVersion}.${parts[2]}`
+}
+
+/**
+ * Get the current version from npm and bump the minor part by 1.
+ */
+async function getNewVersion() {
+  const sharedPackage = core.getInput('shared-package');
+  const result = await exec(`npm view ${sharedPackage} version`);
+  const newVersion = bumpMinorVersion(result.stdout);
+  return newVersion;
+}
+
+async function main() {
+  if (await haveChangesOccurred()) {
+    // eslint-disable-next-line no-console
+    console.log('Changes detected in frontend-shared/, publishing new version...');
+    const newVersion = await getNewVersion();
+    core.setOutput("updated_version", newVersion);
+  } else {
+    // eslint-disable-next-line no-console
+    console.log('No changes detected in frontend-shared/');
+    core.setOutput("updated_version", null);
+  }
+}
+try {
+  main();
```

Comment:
`main` is an async function so you need to use `main().catch(...)` rather than try..catch here.

Reply:
ahh, yes I forgot to clean that up, the try / catch was from an example that I never refactored after I added async bits.
good catch.

Reply:
parseInt(T10)  will be NaN but parseInt(10T) turns into 10 so I think it just lobs off letters without complaining.  Probably not big of a deal.